spec/ruby/core/kernel/caller_spec.rb

```diff
@@ -11,9 +11,10 @@
   end
 
   it 'returns an Array of caller locations using a custom offset' do
-    locations = KernelSpecs::CallerTest.locations(2)
+    locations = KernelSpecs::CallerTest.locations
+    locations_offset = KernelSpecs::CallerTest.locations(2)
 
-    locations[0].should =~ %r{runner/mspec.rb}
+    locations[1..-1].should == locations_offset
```

Comment:
Did it not pass due to the bug mentioned in my first comment?

Reply:
`caller(1..-1)` is not the same as `caller(2)`, please check with MRI when changing a spec expectation.

Reply:
I originally thought that it was failing because of showing core paths but it is a different issue. We have `"(core):1:in `instance_eval'"` for an internal call, MRI reuses previous file and line so it has `".../runner/mspec.rb:101:in `instance_eval'"` which makes the test fail. I've changed the test to use 3. line which is not an internal call and therefore has `runner/mspec.rb`. Having `(core)` instead of bad path seems better to me so I am not going to investigating it further. 

> `caller(1..-1)` is not the same as `caller(2)` ...

That is `Array#[]` method not `caller`. 

I'll change the test back to just use the 3. line of backtrace instead of 2.

Reply:
Sounds good.
The `(core)` part is a bug of Thread::Backtrace::Location#inspect I think.

Reply:
Fix in https://github.com/graalvm/truffleruby/pull/495

Reply:
Ah thanks for the fix. I'll rebase.